Xi/xipassivegrab.c

```diff
@@ -234,31 +229,22 @@ ProcXIPassiveGrabDevice(ClientPtr client)
         }
 
         if (status != GrabSuccess) {
-            xXIGrabModifierInfo *info = modifiers_failed + rep.num_modifiers;
-
-            info->status = status;
-            info->modifiers = *modifiers;
-            if (client->swapped)
-                swapl(&info->modifiers);
+            /* write xXIGrabModifierInfo */
+            x_rpcbuf_write_CARD32(&rpcbuf, *modifiers);
+            x_rpcbuf_write_CARD8(&rpcbuf, status);
+            x_rpcbuf_write_CARD8(&rpcbuf, 0); /* pad0 */
+            x_rpcbuf_write_CARD16(&rpcbuf, 0); /* pad1 */
 
             rep.num_modifiers++;
-            rep.length += bytes_to_int32(sizeof(xXIGrabModifierInfo));
         }
     }
 
-    uint32_t length = rep.length; /* save it before swapping */
-
-    if (client->swapped) {
-        swaps(&rep.sequenceNumber);
-        swapl(&rep.length);
-        swaps(&rep.num_modifiers);
-    }
-    WriteToClient(client, sizeof(rep), &rep);
-    WriteToClient(client, length * 4, modifiers_failed);
+    xi2mask_free(&mask.xi2mask);
+    return X_SEND_REPLY_WITH_RPCBUF(client, rep, rpcbuf);
 
- out:
-    free(modifiers_failed);
+out:
     xi2mask_free(&mask.xi2mask);
+    x_rpcbuf_clear(&rpcbuf);
```

Comment:
We have `goto out` from places before `rpcbuf` declaration. Maybe move the `rpcbuf` declaration to the very top?